### ISSUE TITLE
Factory policies.create=peruser

### ISSUE THREAD

**Issue by skabashnyuk** [eclipse-che/che#14355]
### Is your task related to a problem? Please describe.
Provide an ability to create workspaces from factories per user.

### Describe the solution you'd like
The idea is to have a query parameter **policies.create=peruser** like http://host/f?url=blah&policies.create=peruser that will cause only one workspace to be generated for each unique user that clicks on the factory URL. If the workspace has previously been generated, the existing workspace is reopened.

### Describe alternatives you've considered
none

### Additional context
Che6 persisted factory feature https://www.eclipse.org/che/docs/che-6/factories_json_reference.html#policies

**Comment by tolusha** [comment]
may be `create`?

**Comment by sunix** [comment]
Has that issue been 'triaged' ?

**Comment by sunix** [comment]
@skabashnyuk Is it a issue for platform team ? or do we have anything to do in the Dashboard or che-theia side

**Comment by skabashnyuk** [comment]
@sunix this task is for the team who has more expertise in the dashboard. I think it is ide2.

**Comment by skabashnyuk** [comment]
As I remember I've discussed this task with @evidolob, I thought he wants to delegate it to @olexii4 
Basic idea is to handle this logic on the Dashboard side. 
- Check if  policies.create=peruser set
- if yes -> iterate over all workspaces searching for attribute "factoryurl" = to current query URL
  - if found redirect to that workspace
  - if not create a new workspace and set "factoryurl" attribute.

**Comment by sunix** [comment]
I thought factory service wasn't aware of the dashboard ? @olexii4  ?

**Comment by vitaliy-guliy** [comment]
@skabashnyuk and me have a discussion about workspace creation flow using a factory.
 
http://host/f?url=blah&policies.create=peruser just redirects on http://host//dashboard/#/load-factory/?url=blah&policies.create=peruser and then dashboard creates a workspace using factory and workspace services.

@sunix yes, it looks IDE2 team is responsible for this issue

To have the issue done we need
- ensure that query parameter `policies.create=peruser` is passed to the dashboard
- add a property to the workspace settings/attributes that indicates from which URI this workspace has been created
- while handling a factory in dashboard, find for already existing workspace if query parameter `policies.create=peruser` is set

**Comment by sunix** [comment]
ok

**Comment by olexii4** [comment]
@skabashnyuk  It is very clear and query parameter policies.create=peruser is passed to the dashboard.
But, after creating a new workspace from the devfile with an URL like http://host/f?url=blah (where 'blah' is  an URL for a devfile)
that will create a new workspace with attributes.factoryId = 'undefined'  because any factories were not created at all.  
If I am not mistaken, for nowadays we cannot create a new factory based on a devfile.

Am I right?

And I try to understand how I can find a needed workspace.
WDYT?

**Comment by skabashnyuk** [comment]
@olexii4 I'm not sure I understand your question. Can you rephrase? 
My proposal was https://github.com/eclipse/che/issues/14355#issuecomment-527848916.
```attributes.factoryId``` this is for old Che6 persisted factories. It's not our case.

**Comment by olexii4** [comment]
@skabashnyuk I just confused with the word 'factory'. I thought about CHE factories.
 All are clear now.
